### PR TITLE
[SPLIB] ADD an onDone parameter on animation + related example

### PR DESCRIPTION
**Pull request by damienmarchal**
This is useful to start a new animation when the current one is terminated.

### CODE REVIEW

**Comment by EulalieCoevoet** [comment]
I don't understand what it does... :(
Could you maybe add more info about the parameters here:

`def animate(cb, params, duration, mode="once", onDone=None)`

File: python/splib/animation/animate.py [hunk 1]
```diff
@@ -69,12 +73,14 @@ def onBeginAnimationStep(self,dt):
                 animation.direction = animation.direction
                 animation.startTime = None
                 nextanimations.append(animation)
+            elif animation.onDone is not None:
```
Comment (damienmarchal) [inline python/splib/animation/animate.py:100]: Funny that the indentation is working.

**Comment by damienmarchal** [comment]
Hi @EulalieCoevoet 

I just updated the PR by adding documentation and pep8 cosmetic cleaning.

File: python/splib/animation/animate.py [hunk 2]
```diff
@@ -1,41 +1,69 @@
...
 
 
 class Animation(object):
-    def __init__(self, duration, mode, cb, params):
-        if 'startTime' in params :
+    """An animation clip that trigger callback at regular intervales for a given duration.
+
+           :param mode: specify who the animation will continue (None, "loop", "pingpong")
+           :param duration: the duration of the animation in seconds.
+           :param cb: callback function called each update.
```
Comment (EulalieCoevoet) [inline python/splib/animation/animate.py]: Shouldn't this param be called onUpdate?

Comment (EulalieCoevoet) [reply python/splib/animation/animate.py]: :param mode: specify HOW the animation

**Comment by EulalieCoevoet** [comment]
Look's fine to me. I really need to test this feature. 
Minor comments, otherwise it can be merged.

**Comment by damienmarchal** [comment]
I made the requested changes (rename cb to onUpdate) and fix the type.

**Pull request merged**